pkg/observable/channel/replay.go

```diff
@@ -0,0 +1,209 @@
+package channel
+
+import (
+	"context"
+	"log"
+	"sync"
+	"time"
+
+	"pocket/pkg/observable"
+)
+
+// TODO_CONSIDERATION: perhaps this should be parameterized.
+
+// replayPartialBufferTimeout is the duration to wait for the replay buffer to
+// accumulate at least 1 value before returning the accumulated values.
+const replayPartialBufferTimeout = 100 * time.Millisecond
+
+var _ observable.ReplayObservable[any] = (*replayObservable[any])(nil)
+
+type replayObservable[V any] struct {
+	// observableInternals[V] is embedded to provide safe read and write access
```

Comment:
This is making me `observableInternals` -> `observableSubscriberList` or something along those lines?